spacy/pipeline/trainable_pipe.pyx

```diff
@@ -55,6 +55,41 @@ cdef class TrainablePipe(Pipe):
         except Exception as e:
             error_handler(self.name, self, [doc], e)
 
+
+    def distill(self,
+               teacher_pipe: Optional["TrainablePipe"],
+               teacher_docs: Iterable["Doc"],
+               student_docs: Iterable["Doc"],
+               *,
+               drop: float=0.0,
+               sgd: Optimizer=None,
+               losses: Optional[Dict[str, float]]=None) -> Dict[str, float]:
+        # By default we require a teacher pipe, but there are downstream
+        # implementations that don't require a pipe.
+        if teacher_pipe is None:
+            raise ValueError(Errors.E3000.format(name=self.name))
+        if losses is None:
+            losses = {}
+        if not hasattr(self, "model") or self.model in (None, True, False):
+            return losses
+        losses.setdefault(self.name, 0.0)
+        if not any(len(doc) for doc in teacher_docs):
+            return losses
+        if not any(len(doc) for doc in student_docs):
+            return losses
+        set_dropout_rate(self.model, drop)
+        for node in teacher_pipe.model.walk():
+            if node.name == "softmax":
+                node.attrs["softmax_normalize"] = True
```

Comment:
> I'm assuming it's safe to change this attribute out-of-band in this way (and not revert it after the `predict` call)?

It was made an attribute for this purpose and I fear we currently don't have a better way. Also, it should do no harm, since normalizing softmax doesn't change argmaxes, etc.

> Don't we also need ensure softmax normalization for the student (looks like it'd raise an exception if it's [turned off during training](https://github.com/explosion/thinc/blob/3574f1632f3ac8de8bbc56c5d8e0a209b9a5aa86/thinc/layers/softmax.py#L80))?

Not needed, it's always enabled with `is_train`, which is set for student pipes, since we are backproping.